kv.go

```diff
@@ -155,11 +158,21 @@ func decodeHook(fromType reflect.Type, toType reflect.Type, data interface{}) (i
 
 			return dataOutput, nil
 		} else if fromType.Kind() == reflect.String {
-			b, err := base64.StdEncoding.DecodeString(data.(string))
+			buffer := bytes.NewBuffer([]byte(data.(string)))
+			reader, err := gzip.NewReader(buffer)
```

Comment:
You could make that a bit more composable (might be overkill for the current codebase though — and I'm really bad at naming functions)

```go
func base64Reader(r io.Reader) (*Reader, error) {
    return base64.NewDecoder(base64.StdEncoding, r)
}

func readByPrecedence(r io.Reader, fs ...func(io.Reader) (*Reader, error)) ([]byte, error) {
    var err error
    for _, f := range fs {
        reader, err := f(r)
        if err == nil {
            return ioutil.ReadAll(reader)
        }
    }
    return nil, err
}

// […]
data := bytes.NewBuffer([]byte(data.(string)))
reaturn readByPrecedence(data, gzip.NewReader, base64Reader)
// […]
```